### PR TITLE
Partner assessment api [PD-2869]

### PR DESCRIPTION
**Pull request by tdphillips**
## What's this PR do?
More backend work to support partner assessments

## Screenshots (if appropriate)
Communication record endpoint
2017-07-21-095416_1337x894_scrot

List endpoint (includes create)
2017-07-19-163246_1351x806_scrot

Detail endpoint
2017-07-19-163344_1359x536_scrot

### CODE REVIEW

**Comment by tdphillips** [comment]
Remaining question: How much of the drf work do we think is built-in functionality?

To put it another way, I don't want to write tests for functionality I didn't write. If anyone thinks something is sufficiently custom, say so and I'll write a test.

File: mypartners/views.py [hunk 1]
```diff
@@ -2513,3 +2531,69 @@ def api_get_workflow_states(request):
...
+        company = get_company_or_404(request)
+        partner_id = request.GET['partner_id']
+        if company.partner_set.filter(pk=partner_id).exists():
+            assessments = PartnerAssessment.objects.filter(partner=partner_id)
+            serializer = PartnerAssessmentSerializer(assessments, many=True)
+            return Response(serializer.data)
+        else:
+            return Response({'partner': [u'Invalid partner']})
+
+    @method_decorator(requires('read assessment'))
```
Comment (koonsies) [inline mypartners/views.py]: This should probably be 'create assessment'.

**Comment by koonsies** [comment]
I think the tests here are sufficient.

**Comment by koonsies** [comment]
This looks good for the retrieval of existing partner assessments. We also need an API endpoint for communication records during a particular time period. These will be used to populate the RecordList in the mockups.

**Comment by tdphillips** [comment]
Are start/end dates required or can one/both be left out?

**Comment by koonsies** [comment]
Both will be provided.

**Comment by tdphillips** [comment]
Could/should that be combined with the aggregate endpoint? We're basically doing the same filter generating the same list of records twice.

**Comment by koonsies** [comment]
Oh, yeah. I edited my comment above to ask that same question. I was torn a bit on the issue of keeping the API clean (i.e., doing only one thing) and multiplying the call. Let's go with a single API endpoint in interest of speed.

**Comment by tdphillips** [comment]
Updated the original post. The old aggregate endpoint now also returns a list of records.

**Comment by tdphillips** [comment]
I just took a second glance and realized how ugly that return value is. I'll work on it.

**Comment by tdphillips** [comment]
Made the communication record endpoint more pretty. It no longer resembles the structure returned by other endpoints, however.

**Comment by koonsies** [comment]
Excellent! Thank you. Could you point this PR to the partner-review-stage-2 branch? These APIs aren't needed until that feature branch.

**Comment by tdphillips** [comment]
This is an ugly diff.

**Comment by koonsies** [comment]
A small nitpick, but DRF puts a link at the top of the API that, when clicked, throws a 500 (it trims off the partner_id). I'm not super, super freaked out by this bug, just wanted to call attention to it in case it's a quick fix.

**Pull request merged**